klib/test.c

```diff
@@ -0,0 +1,24 @@
+#include <klib.h>
+
+#define Z_LEN 1024
+
+int y = 123;
+unsigned long z[Z_LEN];
+
+static int foo(int x)
```

Comment:
There seems to be a problem with dynamic relocations, which don't seem to work for global symbols. For example, if I change the `foo()` function above from static to non-static, the load address of the kernel module is not added to the function pointer, and as a result the kernel crashes when it calls foo(). Also calling a non-static foo() from the module itself results in a crash.

Reply:
Commit https://github.com/nanovms/nanos/commit/1dd6799459b6eaf608ca498dafc40c5b08eae589 has a basic implementation (for x86_64) of dynamic relocations that solves the issue.